### PR TITLE
database: Restrict site-admin affiliated errors

### PR DESCRIPTION
**Pull request by tsenart**
Site admins were so far loading errors for all org owned external services, not only their own and site-level.

This caused additional load in the DB as per this [Slack thread](https://sourcegraph.slack.com/archives/CMBA8F926/p1647518650685719).

## Test plan

Integration test.

---

Jira: CLOUD-278

### CODE REVIEW

File: internal/database/external_services_test.go [hunk 1]
```diff
@@ -1212,6 +1223,36 @@ VALUES ($1,'errored', now(), $2)
...
+	_, err = db.Exec(`
+INSERT INTO external_service_sync_jobs (external_service_id, state, finished_at, failure_message)
+VALUES ($1,'errored', now(), $2)
+`, orgOwned.ID, "org failure")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Assert that site-admin should only get errors for site level external services
+	// or self owned external services.
```
Comment (kopancek) [inline internal/database/external_services_test.go:1239]: Should we add a case when the site-admin is also a member of the organization? In that case, that site-admin should get the error.

File: internal/database/external_services.go [hunk 2]
```diff
@@ -1202,7 +1202,11 @@ FROM external_services es
                    ON es.id = essj.external_service_id
                        AND essj.state IN ('completed', 'errored', 'failed')
                        AND essj.finished_at IS NOT NULL
-WHERE ((es.namespace_user_id = %s) OR (%s AND es.namespace_user_id IS NULL))
+WHERE
+  (
+    (es.namespace_user_id = %s) OR
+    (%s AND es.namespace_user_id IS NULL AND es.namespace_org_id IS NULL)
+  )
```
Comment (kopancek) [inline internal/database/external_services.go:1209]: I think a correct check would be:
```suggestion
WHERE
  (
    (es.namespace_user_id = %s) OR
    (es.namespace_org_id IN (SELECT org_id FROM org_members WHERE user_id = %s)) OR 
    (%s AND es.namespace_user_id IS NULL AND es.namespace_org_id IS NULL)
  )
```

Not sure about the performance/syntax here, maybe doing a JOIN on the org_members table would do what we want as well

**Review by kopancek — CHANGES_REQUESTED** [review]

File: internal/database/external_services.go (hunk 2 above)
Comment (kopancek) [reply internal/database/external_services.go:1209]: But now that I think about it, this can be an improvement for a follow up PR 🤔

**Review by kopancek — APPROVED** [review]

File: internal/database/external_services_test.go (hunk 1 above)
Comment (unknwon) [reply internal/database/external_services_test.go:1239]: >Should we add a case when the site-admin is also a member of the organization? In that case, that site-admin should get the error.

+1.

But this PR seems a pure gain so doesn't feel the need to block on this.

**Review by eseliger — APPROVED** [review]

**Pull request merged**